test/test_env.hpp

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/env.hpp>
+
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
+
+inline bool IsTestRunWith(const char* float_arg)
+{
+    assert(float_arg != nullptr);
+    const auto& s_envVar = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
+    return (s_envVar.compare(float_arg) == 0);
+}
```

Comment:
Seems it does not make much sense and it's just an artifact of moving the test from CMake.
Nobody runs the test like `MIOPEN_TEST_FLOAT_ARG=--float ./bin/mlir_test`, everyone runs it like `./bin/mlir_test --float`.
Moreover all the other tests have missed that kind of check.

Reply:
> Nobody runs the test like `MIOPEN_TEST_FLOAT_ARG=--float ./bin/mlir_test`,

@CAHEK7 , my understanding is current gTests still have dependency on env var, and by default MIOPEN_TEST_FLOAT_ARG is set to --float as shown in test/CMakeLists.txt below. Thus this code snippet. Does this make sense?

https://github.com/ROCm/MIOpen/blob/10250d6bcdaef2767d4886b839f639938d15e9e5/test/CMakeLists.txt#L225

Reply:
Except the fact that nobody is using "MIOPEN_TEST_FLOAT_ARG" env variable except a few freshly converted testcases.
Either all the other testcases are wrong and we must fix them, or `MIOPEN_TEST_FLOAT_ARG` is not a way to switch precision.
```
~/miopen$ git grep MIOPEN_TEST_FLOAT_ARG | grep -e "cpp" -e "\.h"
test/gtest/conv_embed_db.cpp:MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
test/gtest/conv_embed_db.cpp:    const auto& s_envVar = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
test/gtest/conv_hip_igemm_xdlops.cpp:MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
test/gtest/conv_hip_igemm_xdlops.cpp:    const auto& s_envVar = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
test/gtest/conv_igemm_mlir.cpp:MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
test/gtest/conv_igemm_mlir.cpp:    const auto& tmp = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
test/gtest/conv_igemm_mlir_xdlops.cpp:MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
test/gtest/conv_igemm_mlir_xdlops.cpp:    const auto& tmp = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
test/gtest/layernorm.cpp:MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
test/gtest/layernorm.cpp:    const auto& tmp = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
test/gtest/sum.cpp:MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
test/gtest/sum.cpp:    const auto& tmp = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
```

And the most important thing is `set(MIOPEN_TEST_FLOAT_ARG --float)` is not an env variable. It is local cmake variable which is passed to the test as a command line argument.

Reply:
@CAHEK7 @cderb `MIOPEN_TEST_FLOAT_ARG` is an environment variable passed to gtests from ctest. Gtests should obey to its value as follows:
- If it is not set, then run tests for all supported datasets
- If it is set to `--float`, or `--half` etc, then only fp32, or fp16 etc. datasets should be run.

More info at #1843